server/app/controllers/admin/AdminApplicationController.java

```diff
@@ -508,6 +509,7 @@ public Result index(
     return ok(
         applicationListView.render(
             request,
+            getCiviFormProfile(request),
```

Comment:
**question:** What happens if you set this before this return statement, like:

`CiviFormProfile profile = profileUtils.currentUserProfile(request).orElseThrow();..`

And then used that variable here?

Not sure why that would make a difference, but just curious if it does.

Reply:
Browser tests still fail with that change.

Still haven't tracked down the exact issue. I'm thinking it might have something to do with AdminLayout being a Singleton. Now that I'm thinking about it, I think it's kind of brittle to rely on the callers to call setOnlyProgramAdminType() anyway. Thinking through what a better option would be. I would want to make the render() classes take in the CiviForm profile so that the AdminLayout can check the profile itself, but those are overridden methods from the base class BaseHtmlLayout. Thinking of instead at least changing setOnlyProgramAdminType to setCiviFormProfile and then checking the profile status before rendering.

Reply:
Yeah that makes sense! Does getting the profile within the layout work as opposed to passing it through?

Reply:
No unfortunately bc it needs the request to get the profile